### PR TITLE
ACU-493: Display payments info on the payments tab

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview
This PR displays the list of payments for an application.

This continues the work done on displaying the payments tab https://github.com/compucorp/uk.co.compucorp.civiawards/pull/182 but adds the list of activity payments for the application instead of the mock table we previously implemented.

This PR is related to https://github.com/compucorp/uk.co.compucorp.civicase/pull/695

## Before
Screen Shot 2021-02-08 at 4 57 04 PM
No real payment activities are displayed.

### Records list
Screen Shot 2021-02-08 at 5 07 52 PM

### Empty state
Screen Shot 2021-02-09 at 10 52 13 AM

### Table Responsiveness

Screen Shot 2021-02-09 at 9 41 14 AM

* Note: placeholders will be added with the filters.

## Technical Details

* A new `gulp test:watch` task was added. This gulp task makes working with tests easier, specially when doing TDD.
* A new `payments-table` directive was created. This directive loads the payment activities and displays them.
* The list of payments types were added as an Angular constant so we could translate payments values such as `'1'` or `'2'` into `'Stipend/Salary'` or `'Expenses'`.
* Some mock values were added for payments, types, and custom fields. The `karma.conf.js` file was updated so it includes these mock values first regardless of the module.
* The `civiawards__reviews-tab__table-container` class was removed from the reviews tab in favour of `table-responsive` because they accomplish the same purpose and we should favour any solution that is already included in Bootstrap.
* The `civiawards__reviews-tab__empty` was removed from both reviews and payment tabs in favour of `civicase__case-tab__empty`, a generic empty state class for all case tabs.

### CODE REVIEW

File: ang/civiawards-payments-tab/directives/payments-table.directive.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<div class="civiawards__reviews-tab__table-container panel panel-default">
```
Comment (deb1990) [inline ang/civiawards-payments-tab/directives/payments-table.directive.html:1]: If we are using `civiawards__reviews-tab__table-container`, we should give it a more generic name.

File: ang/civiawards-payments-tab/directives/payments-table.directive.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+
```
Comment (deb1990) [inline ang/civiawards-payments-tab/directives/payments-table.directive.js:1]: Please remove Extra line.

**Review by deb1990 — CHANGES_REQUESTED** [review]

**Comment by deb1990** [comment]
@reneolivo Can we also add a Empty State, just like Review Tab?

**Comment by swastikpareek** [comment]
@reneolivo : 
* Should we make the PR title a bit more descriptive. `ACU-493: Display payments info on the payments tab`?
* Is the PR following the [PR writing guidelines](https://compucorp.atlassian.net/wiki/spaces/SD/pages/1520304185/Pull+Requests+guidelines)?

**Review by deb1990 — APPROVED** [review]

**Pull request merged**